src/server/chunk_server/chunk_server_impl.h

```diff
@@ -0,0 +1,95 @@
+#ifndef GFS_SERVER_CHUNK_SERVER_IMPL_H_
+#define GFS_SERVER_CHUNK_SERVER_IMPL_H_
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/time/time.h"
+#include "google/protobuf/stubs/statusor.h"
+#include "grpcpp/grpcpp.h"
+#include "src/common/config_manager.h"
+#include "src/common/protocol_client/chunk_server_service_server_client.h"
+#include "src/common/protocol_client/master_chunk_server_manager_service_client.h"
+#include "src/common/utils.h"
+
+namespace gfs {
+namespace server {
+
+class ChunkServerImpl {
+ public:
+  static google::protobuf::util::StatusOr<ChunkServerImpl*>
+  ConstructChunkServerImpl(const std::string& config_filename,
+                           const std::string& chunk_server_name,
+                           const bool resolve_hostname = false);
+
+  // Add or update the write lease for |file_handle| to expire at UNIX seconds
+  // timestamp of |expiration_unix_sec|.
+  void AddOrUpdateLease(const std::string& file_handle,
+                        const uint64_t expiration_unix_sec);
+  void RemoveLease(const std::string& file_handle);
+  bool HasWriteLease(const std::string& file_handle);
+  // Return NOT_FOUND, if no lease exists for file handle
+  google::protobuf::util::StatusOr<absl::Time> GetLeaseExpirationTime(
+      const std::string& file_handle);
+
+  // MOCK FUNCTIONS for chunk file manager
+  // Initialize, or update the chunk |version| for the given |file_handle|
+  // TODO(tugan,michael): use chunk file manager instead, when ready
+  void SetChunkVersion(const std::string& file_handle, const uint32_t version);
```

Comment:
nit: this will need to pass both the current version and the new version you want to set it to, to the file chunk mgr. We check if the current version of the chunk matches the specified current version before updating to new version. Fine for now.

Reply:
this mock is not used except in test for temp purposes; prob gonna remove in next PR when i integrate your file manager